api/client/src/main/java/org/projectnessie/client/builder/BaseAssignReferenceBuilder.java

```diff
@@ -15,17 +15,19 @@
  */
 package org.projectnessie.client.builder;
 
-import org.projectnessie.client.api.AssignBranchBuilder;
 import org.projectnessie.model.Reference;
 
-public abstract class BaseAssignBranchBuilder extends BaseOnBranchBuilder<AssignBranchBuilder>
-    implements AssignBranchBuilder {
+public abstract class BaseAssignReferenceBuilder<R> extends BaseOnReferenceBuilder<R> {
```

Comment:
Let's keep this class package-private as well

Reply:
discussed in https://github.com/projectnessie/nessie/pull/7348#discussion_r1291601444